velox/exec/GroupingSet.h

```diff
@@ -91,6 +91,13 @@ class GroupingSet {
     return table_ ? table_->numDistinct() : 0;
   }
 
+  /// If there are global grouping sets, then returns if they have default
+  /// output incase no input rows were received.
+  bool hasDefaultGlobalGroupingSetOutput() const {
```

Comment:
Do we need both hasDefaultGlobalGroupingSetOutput and getDefaultGlobalGroupingSetOutput as public method? Why not just have one, getDefaultGlobalGroupingSetOutput, and have it return 'false' if there is no output.

Reply:
So we need to size the output buffer in HashAgg before invoking groupingSet output function. Its cleaner to size it only if there is default aggregation output, else we have to size and then undo it. That's why I kept 2 functions.